native/packages/react-native-bpk-component-input/src/BpkInput.js

```diff
@@ -0,0 +1,172 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  View,
+  TextInput,
+  Image,
+  Platform,
+  StyleSheet,
+} from 'react-native';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const IOS_TOKENS = require('bpk-tokens/tokens/ios/base.react.native.common.js');
+const ANDROID_TOKENS = require('bpk-tokens/tokens/android/base.react.native.common.js');
+
+const tokens = Platform.select({
+  ios: () => IOS_TOKENS,
+  android: () => ANDROID_TOKENS,
+})();
+
+const {
+  colorGray100,
+  colorGray300,
+  colorGray700,
+  colorGreen500,
+  colorRed500,
+  spacingMd,
+  spacingSm,
+  textLgFontSize,
+  textLgFontWeight,
+  textXsFontSize,
+  textXsFontWeight,
+} = tokens;
+
+const styles = StyleSheet.create({
+  input: {
+    borderColor: colorGray100,
+    borderWidth: 1,
+    borderRadius: spacingSm,
+    padding: spacingSm * 3,
+    flexDirection: 'row',
+    height: 48,
+    width: 'auto',
+  },
+  text: {
+    fontSize: textLgFontSize,
+    color: colorGray700,
+    width: 'auto',
+    flex: 1,
+    fontWeight: Platform.OS === 'ios' ? textLgFontWeight : undefined,
+    fontFamily: Platform.OS === 'android' ? 'sans-serif-medium' : 'System',
+  },
+  smallInput: {
+    padding: spacingMd,
+    height: 32,
+  },
+  smallText: {
+    fontSize: textXsFontSize,
+    fontWeight: Platform.OS === 'ios' ? textXsFontWeight : undefined,
+    fontFamily: Platform.OS === 'android' ? 'sans-serif' : 'System',
+  },
+  placeholderText: {
+    fontStyle: 'italic',
+  },
+  disabledText: {
+    color: colorGray100,
+  },
+  icon: {
+    flex: 0,
+    width: 22,
+    height: 22,
+  },
+  validIcon: {
+    tintColor: colorGreen500,
+  },
+  invalidIcon: {
+    tintColor: colorRed500,
+  },
+  smallIcon: {
+    width: 14,
+    height: 14,
+  },
+});
+
+const BpkInput = (props) => {
+  const { small, valid, value, disabled, style: innerStyle, ...rest } = props;
+
+  let iconSource = null;
+
+  const style = [styles.input];
+  const textStyle = [styles.text];
+  const iconStyle = [styles.icon];
+  if (small) {
+    style.push(styles.smallInput);
+    textStyle.push(styles.smallText);
+    iconStyle.push(styles.smallIcon);
+  }
+  if (disabled) {
+    textStyle.push(styles.disabledText);
+  }
+  if (value === '') {
+    textStyle.push(styles.placeholderText);
+  }
+  if (valid !== null) {
+    iconSource = valid ? require('./tick-circle.png') : require('./exclamation-circle.png'); // eslint-disable-line
```

Comment:
Can we add a TODO here to replace with icon once when it's shipped - usually with a link to a card for the work item.